### PR TITLE
Add batch synthesis sample code

### PR DESCRIPTION
**Pull request by Ling-Cao**
## Purpose

* Add C# sample code for TTS batch synthesis API.

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: Add sample code
```

### CODE REVIEW

**Comment by eric-urban** [comment]
@Ling-Cao @dargilco - Should the folder be named "batchsynthesis" or "batch-synthesis" instead of "batchSynthesis"?

File: samples/batchSynthesis/BatchSynthesisSample/dto/BatchSynthesis.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
+
+public class BatchSynthesis : BatchSynthesisDefinition
+{
+    /// <summary>
+    /// Gets or sets the identifier of this batch synthesis.
```
Comment (eric-urban) [inline samples/batchSynthesis/BatchSynthesisSample/dto/BatchSynthesis.cs:9]: Can we cut "Gets or sets" from all descriptions? Focus on what is the property. The get/set is clear from code.

Comment (Ling-Cao) [reply samples/batchSynthesis/BatchSynthesisSample/dto/BatchSynthesis.cs]: Sure, removed "Gets or sets",  not auto generated, there is no swagger file to public yet.

**Comment by Ling-Cao** [comment]
> @Ling-Cao @dargilco - Should the folder be named "batchsynthesis" or "batch-synthesis" instead of "batchSynthesis"? @yulin-li created [batch-synthesis](https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/yulin/batch-synthesis-sample/samples/batch-synthesis) which seems good.

File: samples/batchSynthesis/BatchSynthesisSample/dto/BatchSynthesis.cs (hunk 1 above)
Comment (Ling-Cao) [reply samples/batchSynthesis/BatchSynthesisSample/dto/BatchSynthesis.cs]: I refined the folder name and structure to align with Python sample code which looks good for me.

**Comment by Ling-Cao** [comment]
> > @Ling-Cao @dargilco - Should the folder be named "batchsynthesis" or "batch-synthesis" instead of "batchSynthesis"? @yulin-li created [batch-synthesis](https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/yulin/batch-synthesis-sample/samples/batch-synthesis) which seems good.

I refined the folder name and structure to align with Python sample code which looks good for me.

**Comment by yulin-li** [comment]
close, see #1730

**Pull request closed without merging**